src/types/ExchangeApi.ts

```diff
@@ -0,0 +1,36 @@
+import BigNumber from 'bignumber.js';
+
+export interface ExchangeApi {
+    getApiTokenName(tokenSymbol: string): string;
+
+    queryExchangeRate(
+        sourceCurrency: string,
+        targetCurrency: string
+    ): Promise<BigNumber>;
+}
+
+export abstract class BaseExchangeApi {
```

Comment:
I find it a bit confusing to put a class inside a /types folder. Would move it next to the other ExchangeApis.

Reply:
updated